enricher/eks/metric.go

```diff
@@ -0,0 +1,89 @@
+package eks
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/canva/amazon-kinesis-streams-for-fluent-bit/enricher/mappings"
+	"github.com/canva/amazon-kinesis-streams-for-fluent-bit/metricserver"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+)
+
+type EnricherMetric struct {
+	meter metric.Meter
+
+	outputRecordCount  metric.Int64Counter
+	outputDroppedCount metric.Int64Counter
+}
+
+func WithMetricServer(ms *metricserver.MetricServer) EnricherConfiguration {
+	return func(e *Enricher) error {
+		meter := ms.GetMeter("github.com/canva/amazon-kinesis-streams-for-fluent-bit/enricher/eks")
+		outputRecordCount, err := meter.Int64Counter("fluentbit.output.record", metric.WithDescription("output record counter"))
+
+		if err != nil {
+			return err
+		}
+
+		outputDroppedCount, err := meter.Int64Counter("fluentbit.output.dropped.body", metric.WithDescription("output dropped record counter where log or message does not exist"))
+
+		if err != nil {
+			return err
+		}
+
+		e.metric = &EnricherMetric{
+			meter:              ms.GetMeter("github.com/canva/amazon-kinesis-streams-for-fluent-bit/enricher/eks"),
+			outputRecordCount:  outputRecordCount,
+			outputDroppedCount: outputDroppedCount,
+		}
+
+		return nil
+	}
+}
+
+func (e *Enricher) AddRecordCount(record map[interface{}]interface{}, logType LogType) {
+	if e.metric == nil {
+		return
+	}
+
+	var serviceName = inferServiceName(record, logType)
+
+	e.metric.outputRecordCount.Add(context.Background(), 1, metric.WithAttributes(attribute.Key(mappings.RESOURCE_SERVICE_NAME).String(serviceName)))
+}
+
+func (e *Enricher) AddDropCount() {
+	if e.metric == nil {
+		return
+	}
+
+	e.metric.outputDroppedCount.Add(context.Background(), 1)
+}
+
+func inferServiceName(record map[interface{}]interface{}, logType LogType) string {
+	// fallback in case unable to find service name.
+	var serviceName interface{} = "_missing"
```

Comment:
Lets use mappings.PLACEHOLDER_MISSING_KUBERNETES_METADATA for consistency with the rest of our pipeline

Reply:
@mallyx3 I had some more things on this, I think we should use another label, the reason is if it fails to find service name, this shouldn't pollute `PLACEHOLDER_MISSING_KUBERNETES_METADATA`. WDYT?

Reply:
Discussed offline. Did not want to polute `PLACEHOLDER_MISSING_KUBERNETES_METADATA` so defaulting to `_missing_service_name`